### PR TITLE
Include Secrets Provider init app in E2E workflow

### PR DESCRIPTION
**Pull request by john-odonnell**
### What does this PR do?
In #272, the app deploy Helm Chart was updated to support deployment of the app + Secrets Provider init container.
Here, the E2E workflow scripts are updated to deploy and test against this application.
Also, updates the app + Secrets Provider init container subchart for Openshift deployment, similar to changes to the `app-summon-sidecar` and `app-secretless-broker` subcharts made in PR #346.

### What ticket does this PR close?
Resolves #[relevant GitHub issues, eg 76]

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

File: bin/test-workflow/8_app_verify_authentication.sh [hunk 1]
```diff
@@ -104,6 +104,7 @@ else
   init_url_with_host_outside_apps="test-app-with-host-outside-apps-branch-summon-init.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
   sidecar_url="test-app-summon-sidecar.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
   secretless_url="test-app-secretless-broker.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
+  secrets_provider_init_url="test-app-secrets-provider-init.$TEST_APP_NAMESPACE_NAME.svc.cluster.local:8080"
```
Comment (diverdane) [inline bin/test-workflow/8_app_verify_authentication.sh:107]: We'll need a `secrets_provider_init_url` definition for `"$PLATFORM" == openshift" case above (Line 99).
We shouldn't need the `init_url` (Line 97) since that's not in our test matrix, and it's probably going to get deprecated real soon. So feel free to use port `8082` for `secrets_provider_init_url`.

**Review by diverdane — COMMENTED** [review]
LGTM, except for missing port def.

File: bin/test-workflow/8_app_verify_authentication.sh (hunk 1 above)
Comment (john-odonnell) [reply bin/test-workflow/8_app_verify_authentication.sh:107]: Sounds good.
Does this deprecation also apply to `init_url_with_host_outside_apps`?

Comment (diverdane) [reply bin/test-workflow/8_app_verify_authentication.sh:107]: Interesting question. I thought about bringing this up earlier, but I think this merits a discussion on whether this can be deleted or not. My thinking is that we should consider deleting this `init_url_with_host_outside_apps` test case. This test case is a bit of a corner case that's more about testing functionality of Conjur and Conjur security policy. All of the other application test cases use an `apps` policy group... and they require that the application Pod matches a specified Namespace, Deployment, ServiceAccount, and authn container name. For the `init_url_with_host_outside_apps`, the authn-k8s container uses a host ID that isn't part of the `apps` group, and it isn't as particular about checking Deployment and ServiceAccount.

For the other `secrets_provider_init_url` application, I looked back at the discussion for using `authn-k8s` as an init container. Apparently the consensus is not to deprecate, but to advise against using authn-k8s as an in it container  in production (i.e. advise using authn-k8s as a sidecar instead in production).

Comment (john-odonnell) [reply bin/test-workflow/8_app_verify_authentication.sh:117]: Per our discussion - authn-k8s init container is not deprecated, but advised against in production - `init` URLs are maintained.

**Review by diverdane — APPROVED** [review]
Thanks for getting this workflow in!

**Review by diverdane — APPROVED** [review]

**Comment by diverdane** [comment]
Thanks for adding the template fix!
